[muhdbhz] iP

muhdbhz opened the pull request:
### Moody

> “You never finish a program, you just stop working on it” – Anonymous ([source](https://softwarequotes.com/quote/you-never-finish-a-program--you-just-stop-working-))

Moody is here to simplify your task management and make life much easier! 

It has several key features: 

- Text-based so typing commands is straightforward
- Easy to pick up
- INCREDIBLY fast to use

All you need to do is:

1. download it from [here](https://github.com/muhdbhz/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks (todo, deadlines and events)
- [x] Search functions
- [x] Marking/Unmarking of tasks
- [ ] Modern GUI (coming soon)

Example code for the main method::
```java
public static void main(String[] args) {
        new Moody("./data/moody.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

muhdbhz committed:
Rename Duke.java to Moody.java

muhdbhz committed:
Add the ability to echo user's input

muhdbhz committed:
Add the ability to store and return text by user

muhdbhz committed:
Add the ability to mark and unmark tasks status

Create a task class

muhdbhz committed:
Add support for tracking three types of tasks

Create a deadline class
Create an event class
Create a todo class

muhdbhz committed:
Use the input/output redirection technique to semi-automate the testing of Duke

muhdbhz committed:
Add exception handling for incorrect inputs by user

Create a invalidcommandexception class
Create a taskinputexception class
Create a taskoutofboundsexception class

muhdbhz committed:
Add support for deleting tasks from the list

muhdbhz committed:
Add the ability to save tasks

muhdbhz committed:
Merge branch 'branch-Level-7'

muhdbhz committed:
Add the ability to understand date and time

muhdbhz committed:
Merge branch 'branch-Level-8'

muhdbhz committed:
Refactor closely related code as classes

muhdbhz committed:
Merge branch 'branch-A-MoreOOP'

muhdbhz committed:
Organize classes into packages

muhdbhz committed:
Merge branch 'branch-A-Packages'

muhdbhz committed:
Merge remote-tracking branch 'origin/add-gradle-support'

muhdbhz committed:
Add two test classes

muhdbhz committed:
Include parent package moody for existing packages

muhdbhz committed:
Merge branch 'branch-A-Jar'

muhdbhz committed:
Add javadoc comments for most classes

muhdbhz committed:
Change the code to comply with coding standards

muhdbhz committed:
Add the ability to find tasks

muhdbhz committed:
Merge branch 'branch-A-JavaDoc'

muhdbhz committed:
Merge branch 'branch-A-CodingStandard'

muhdbhz committed:
Resolve merge conflicts and enable finding tasks

muhdbhz committed:
Resolve merge conflicts and enable finding tasks

muhdbhz committed:
Fix errors in code

hansneddyanto commented on src/main/java/moody/Moody.java:
```diff
@@ -0,0 +1,71 @@
+ * interacting with storage, and processing user commands.
+ */
+public class Moody {
+    private final Storage storage;
+    private final TaskList tasks;
+    private final Ui ui;
```
You might want to CAPS LOCK final variables

muhdbhz replied:
Actually do I have to do that for private final fields as well? My understanding was to do it on private static final fields... I might be wrong. Could I check with you on that?

hansneddyanto commented on src/main/java/moody/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    public AddDeadlineCommand(String description, String deadline) throws DateTimeParseException {
+        this.description = description;
+        this.deadline = LocalDateTime.parse(deadline, INPUT_FORMATTER);
```
Maybe to avoid confusion with class Deadline, you might want to rename this (maybe to 'dateBy')?

hansneddyanto commented on src/main/java/moody/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+
+import moody.exceptions.InvalidCommandException;
+import moody.exceptions.TaskInputException;
+import moody.commands.*;
```
Do not use wildcards

hansneddyanto commented on src/main/java/moody/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+        String command = parts[0].toLowerCase(); // Get the command keyword
+        String arguments = parts.length > 1 ? parts[1] : ""; // Get the arguments
+
+        switch (command) {
```
Good job on correct indentation for switch case

hansneddyanto commented on src/main/java/moody/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+    public static Command parse(String input) throws InvalidCommandException, TaskInputException {
+        String[] parts = input.split(" ", 2); // Split into command and arguments
+
+        String command = parts[0].toLowerCase(); // Get the command keyword
```
To avoid confusion with Command class, you might want to change the variable name to 'commandWord' instead

hansneddyanto commented on src/main/java/moody/storage/Storage.java:
```diff
@@ -0,0 +1,103 @@
+            boolean isTaskDone = parts[1].equals("1");
+            Task task;
+
+            switch (taskType) {
+                case "T":
+                    task = new Todo(parts[2].trim());
+                    break;
+                case "D":
+                    task = new Deadline(parts[2].trim(), LocalDateTime.parse(parts[3].trim(), INPUT_FORMATTER));
+                    break;
+                case "E":
+                    task = new Event(parts[2].trim(), LocalDateTime.parse(parts[3].trim(), INPUT_FORMATTER),
+                            LocalDateTime.parse(parts[4].trim(), INPUT_FORMATTER));
+                    break;
+                default:
+                    continue;
+            }
```
There is no indentation for the switch case

hansneddyanto commented on src/main/java/moody/tasks/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+ * A Deadline is a type of Task that includes a date and time by which the task is due.
+ */
+public class Deadline extends Task {
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy h:mm a");
```
Good job on CAPS LOCK for final variables

hansneddyanto reviewed (commented):
Overall naming conventions are ok. LGTM!

lithops-zty commented on src/main/java/moody/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,65 @@
+    @Override
+    public boolean isExit() {
+        return false;
+    }
```
Since this is the default implementation, perhaps this can be removed (same for other non-exit commands)?
```suggestion
```

lithops-zty commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# moody.Moody project template
```
Consider removing (possibly IntelliJ refactor auto-generated) `moody.`
```suggestion
# Moody project template
```

lithops-zty commented on src/main/java/moody/exceptions/TaskInputException.java:
```diff
@@ -0,0 +1,8 @@
+package moody.exceptions;
+
+public class TaskInputException  extends Exception {
```
extra whitespaces
```suggestion
public class TaskInputException extends Exception {
```

lithops-zty commented on src/main/java/moody/ui/Ui.java:
```diff
@@ -0,0 +1,151 @@
+     * Displays a horizontal line to separate sections of text.
+     * Prints a spacer line to create visual separation in the output.
+     */
+    public void showLine() {
+        System.out.println(LINE);
+    }
```
I think it may be better to use this method to print divider lines in other methods defined in this class.

lithops-zty reviewed (commented):
Overall LGTM! Just some minor suggestions you can consider. JavaDoc is very well written; well-organized code using packages and inheritance. Good job!

muhdbhz committed:
Add a GUI to Moody chatbot

muhdbhz committed:
Merge branch 'branch-Level-10'

muhdbhz committed:
Add assertions to code

muhdbhz committed:
Refactor parsing methods and add Javadoc documentation

- Consolidate task index parsing into a single method to eliminate repetitive code.
- Add Javadoc comments to methods for consistent and clear documentation.

muhdbhz committed:
Merge pull request #1 from muhdbhz/branch-A-Assertions

Add assertions to code

muhdbhz committed:
Merge pull request #2 from muhdbhz/branch-A-CodeQuality

Refactor parsing methods and add Javadoc documentation

muhdbhz committed:
Add tag and untag command parsing for improved task management

Previously, there was no support for tagging tasks in the CLI. Adding tag and untag functionality enhances the ability to categorize and manage tasks effectively.

Let's implement parseTagCommand and parseUntagCommand to allow users to tag tasks with descriptive labels, improving task organization and retrieval.

Let's also ensure that the new tag and untag commands are compatible with existing command parsing logic and provide appropriate error messages for invalid inputs.

muhdbhz committed:
Merge pull request #3 from muhdbhz/branch-C-Tagging

Add tag and untag command parsing for improved task management

muhdbhz committed:
Use ChatGPT to assist with code quality

Certain classes in the code lack documentation.

As such, we use ChatGPT to assist with JavaDoc documentation.

muhdbhz committed:
Merge pull request #5 from muhdbhz/branch-A-AiAssisted

Use ChatGPT to assist with code quality

muhdbhz committed:
Fix the Tag Class and update README

1. Current Tag Class does not show the intended output in the GUI.

2. README is not user-friendly.

We fix bugs to display the correct output, using streams.
We also update the README for user-friendliness.

muhdbhz committed:
Merge pull request #6 from muhdbhz/branch-A-UserGuide

Fix the Tag Class and update README

muhdbhz committed:
Fix bugs in Storage

muhdbhz committed:
Fix build.gradle not working for fat JAR

muhdbhz committed:
Fix README not displaying properly

muhdbhz committed:
Fix README not displaying picture
